src/App.tsx

```diff
@@ -1,14 +1,50 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import { getTodos } from './api';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodoId, setSelectedTodoId] = useState<number>(0);
+  const [filter, setFilter] = useState('All');
+  const [query, setQuery] = useState('');
+
+  const selectedTodo = todos.find(todo => todo.id === selectedTodoId);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos);
+  }, []);
+
+  const closeModal = useCallback(() => (
+    setSelectedTodoId(0)
+  ), []);
+
+  const visibleTodos = todos.filter(todo => {
```

Comment:
It would be great if we don't filter the array, if no query passed

Reply:
Also, you could try to apply a debounce for filtering todos by query (it's not required, but it would be fun)